torch/_inductor/select_algorithm.py

```diff
@@ -86,10 +86,14 @@ def __init__(self, code, replacement_hooks):
         self.code = code
         self.replacement_hooks = replacement_hooks
 
-    def finalize_hook(self, hook_key: str) -> None:
-        assert (
-            hook_key in self.replacement_hooks
-        ), f"{hook_key} not registered in self.replacement_hooks"
+    def finalize_hook(self, hook_key: str, strict=True) -> None:
```

Comment:
nit: I wish I knew why we need strict = False

Reply:
Oh, it's that the other hooks are run on every single template. But in this case, not all templates have `gen_argdefs` on them.